tests/std/tests/P0040R3_extending_memory_management_tools/test.cpp

```diff
@@ -196,6 +196,120 @@ void test_destroy_n() {
     assert(g_alive == 0);
 }
 
+struct copy_elision_dest;
+
+class pinned {
+public:
+    explicit pinned(int n) : n_{n} {}
+
+    pinned(const pinned&)            = delete;
+    pinned& operator=(const pinned&) = delete;
+
+private:
+    friend copy_elision_dest;
+
+    int n_;
+};
+
+class pinned_ioterator {
+private:
+    struct arrow_proxy {
+        pinned val_;
+
+        pinned* operator->() {
+            return &val_;
+        }
+    };
+
+public:
+    using value_type        = pinned;
+    using difference_type   = int;
+    using reference         = pinned;
+    using pointer           = arrow_proxy;
+    using iterator_category = std::input_iterator_tag;

```

Comment:
We conventionally try to follow the Standard's canonical ordering for the nested iterator types:
```suggestion
    using iterator_category = std::input_iterator_tag;
    using difference_type   = int;
    using value_type        = pinned;
    using pointer           = arrow_proxy;
    using reference         = pinned;
```